### ISSUE TITLE
About the learning rate  setting of p_conv and m_conv 

### ISSUE THREAD

**Issue by dontLoveBugs** [4uiiurz1/pytorch-deform-conv-v2#7]

**Comment by 4uiiurz1** [comment]
You're right!
I'll fix it.

**Comment by BananaLv26** [comment]
> You're right!
> I'll fix it.

Have you solved this problem now?

**Comment by jszgz** [comment]
@dontLoveBugs Hello, can you review my [issue ](https://github.com/4uiiurz1/pytorch-deform-conv-v2/issues/21)? I think the bilinear kernel is wrong

**Comment by zcong17huang** [comment]
> You're right!
> I'll fix it.

'tuple' object can not be modified. Your code just get an generator.

**Comment by XinZhangRadar** [comment]
I have searched online, the  grad of output can not be modified, if you want modify the grad of input, you need to return the modified grad of input , like :
  def _set_lr(module, grad_input, grad_output):
         return  (grad_input[i] * 0.1 for i in range(len(grad_input)))

you can try it. My question is ：
Why change the p_conv gradients, Is it to avoid affecting the learning of another feature extraction branch?

**Comment by steven22tom** [comment]
@XinZhangNLPR the you is becuse the backforward_hook expected tuple, not  'generator'

> I have searched online, the grad of output can not be modified, if you want modify the grad of input, you need to return the modified grad of input , like :
> def _set_lr(module, grad_input, grad_output):
> return (grad_input[i] * 0.1 for i in range(len(grad_input)))
> 
> you can try it. My question is ：
> Why change the p_conv gradients, Is it to avoid affecting the learning of another feature extraction branch?

Your suggestion still return a generator not a tuple

**Comment by YXB-NKU** [comment]
> You're right! I'll fix it.

it seems this bug has not fixed yet
